src/app/components/dataset-properties-modal/dataset-properties-modal.component.ts

```diff
@@ -38,7 +38,10 @@ export class DatasetPropertiesModal implements OnInit {
     @Inject(MAT_DIALOG_DATA) data,
   ) 
   {
-    const node = data.event;
+    let node = data.event;
+    if(node && !node.data) {
+      node = { data: { datasetAttrs: JSON.parse(JSON.stringify(node)), fileName: node.fullName}}
```

Comment:
We already check if node.data a few lines below, so that conditional should be worked into an "else"